### ISSUE TITLE
CMake > make

### ISSUE THREAD

**Issue by Commentsareuseless** [ArturMarekNowak/Networking#1]
Consider using CMake instead of manually creating make files. CMake supports C# projects since version 3.8.2 ([docs](https://cmake.org/cmake/help/v3.8/release/3.8.html)) (This would make project look more professional ;) )

**Comment by ArturMarekNowak** [comment]
Thanks mate, I will gladly try to utilize CMake in this project :D I was not aware that the CMake supports C# in any way. I was always making my own Makefile for sake of my own practice and simplicity. But I will definitely give it a shot :D
![pobrane](https://user-images.githubusercontent.com/60242009/134822341-64785b66-1047-4558-93ba-e5224c30b4c1.jpg)

**Comment by ArturMarekNowak** [comment]
I took a look into C# and CMake combination. More specificly I have decided to use this github repository https://github.com/bemehiser/cmake_csharp_example and oh boy that's a lot of files generated out there. I have decided to forfeit the idea of making Makefiles and instead I have made multiproject solutions in Jetbrains Rider which tbh I should have done in the first place. More over I have stumbled upon this stack overflow thread in which user [Max Ehrlich](https://stackoverflow.com/users/564540/max-ehrlich) states that usage of CMake and C# is not so good since C# is cross platform by design unless some wild hybrid of C++ and C# projects is made. But thank you for your advise. That's a good tool right there :D 

Source: https://stackoverflow.com/questions/2074144/generate-c-sharp-project-using-cmake/55602125#55602125
